src/window/windows/castlewindow_winapi.inc

```diff
@@ -1422,8 +1422,8 @@ function TCastleWindowBase.BackendFileDialog(const Title: string; var FileName:
   end;
 
 var
-  Struct: TOpenFileName;
-  FileNameBuffer: array [0..6000] of char;
+  Struct: TOpenFileNameW;
+  FileNameBuffer: array [0..Pred(MAX_PATH+10)] of widechar;
```

Comment:
I see you change 6000 into `Pred(MAX_PATH+10)`. What is the reason?

MAX_PATH is 260 (but you can have longer paths, and ideally everything works with it, if you adjust Windows registry settings --- https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd ). So our 6000 was allowing to pass longer paths.

Granted, hardcoding 6000 is ugly :) But then, `Pred(MAX_PATH+10)` means we just hardcode a smaller number, because file paths may still be longer than MAX_PATH.

Reply:
I didn't know that Windows10 allowed over 256 path lengths.
Therefore, I restored the path length limitation.